test/website/test.js

```diff
@@ -0,0 +1,141 @@
+const blc     = require('broken-link-checker');
+const http    = require('http');
+const https   = require('https');
+const util    = require('gulp-util');
+const url     = require('url');
+const parse5  = require('parse5');
+const Promise = require('pinkie');
+const walk    = require('dom-walk');
+
+
+class WebsiteTester {
+    constructor () {
+        this.parsedDocs    = {};
+        this.checkedHashes = {};
+    }
+
+    _hasHash (tree, hash) {
+        const attrMatches = attr => (attr.name === 'name' || attr.name === 'id') && attr.value === hash;
+        const hasId       = node => node.attrs && node.attrs.some(attrMatches);
+
+        return walk(tree, hasId);
+    }
+
+    _parsePage (pageUrl) {
+        if (this.parsedDocs[pageUrl])
+            return Promise.resolve(this.parsedDocs[pageUrl]);
+
+        return new Promise(resolve => {
+            const protocol = url.parse(pageUrl).protocol === 'http:' ? http : https;
+
+            protocol.get(pageUrl, resolve);
+        })
+        .then(response => {
+            return new Promise(resolve => {
+                let data = '';
+
+                response.on('data', chunk => data += chunk);
+                response.on('end', () => resolve(data));
+            });
+        })
+        .then(body => {
+            const parser   = new parse5.Parser();
+            const document = parser.parse(body);
+
+            this.parsedDocs[pageUrl] = document;
+            return document;
+        });
+    }
+
+    _isHashValid (link) {
+        const parsedLink   = url.parse(link);
+        const hashlessLink = link.substring(0, link.indexOf('#'));
+
+        if (this.checkedHashes[link])
+            return Promise.resolve(this.checkedHashes[link]);
+
+        return Promise
+            .resolve(this._parsePage(hashlessLink))
```

Comment:
No need for `resolve` here, since `_parsePage` is already async you can just attach `.then` to it